td/telegram/StickersManager.cpp

```diff
@@ -1326,7 +1326,7 @@ tl_object_ptr<telegram_api::InputMedia> StickersManager::get_input_media(
   }
   CHECK(!file_view.has_remote_location());
 
-  if (input_file != nullptr) {
+  if (input_file != nullptr && input_thumbnail != nullptr) {
```

Comment:
You've just disabled possibility to send stickers without thumbnail which is perfectly valid.

Reply:
@levlam if it's valid, so how can this be correct?

```
inputMediaUploadedDocument>(.... std::move(input_file), std::move(input_thumbnail) .... )
```
```input_thumbnail``` can be null

Reply:
input_thumbnail has type std::unique_ptr<telegram_api::InputFile>, and so it can be equal to nullptr.
In the mentioned line std::unique_ptr is moved as parameter to telegram_api::inputMediaUploadedDocument constructor through function similar to make_unique, which is legal for std::unique_ptr (and it is also legal for raw pointers), because there is no pointer dereference anywhere.